Sources/Apodini/Request/Request.swift

```diff
@@ -5,27 +5,41 @@
 //  Created by Paul Schmiedmayer on 7/12/20.
 //
 
-import Vapor
+import Foundation
+import Fluent
+
+
+public class Request {
```

Comment:
Since Requests can look very different for different protocols. 
Perhaps we should go with a protocol:

```
protocol Request {
    func parameter<T : InputType>(for uuid: UUID) throws -> T
    ... other requirements 
}
```

@PSchmiedmayer what do you think if we add an httpHeader option to parameter?

Reply:
It would be easier to do this:  `func parameter<T : Codable>(for parameter: Parameter<T>) throws -> T`.

Reply:
the part with `LosslessStringConvertible` is still a bit ugly though.